### ISSUE TITLE
DFRN: Connection posts don't show the full link, fails the protocol whitelist

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#3412]
All the connection post I receive (A is now friends with B) have B's link wrong and shows as invalid.

See: https://friendica.mrpetovan.com/display/00000000e4a2ac315905f057ebdde44aea9ed342

I tried to look into it, but I'm confused, here's an excerpt of the item row:

Fiel | Value
------------ | -------------
guid | 00000000e4a2ac315905f057ebdde44aea9ed342
verb | http://activitystrea.ms/schema/1.0/make-friend
object-type | http://activitystrea.ms/schema/1.0/person
object | See below
body | `is now friends with`
rendered-html | `<a href="https://f.haeder.net/profile/roland?zrl=https%3A%2F%2Ffriendica.mrpetovan.com%2Fprofile%2Fhypolite" target="_blank">Roland H&auml;der</a> is now friends with <a href="javascript:void(0)" target="_blank" class="invalid-href" title="Invalid link protocol">cb7f604332cf39</a>`

XML Object
````xml
<object>
  <type>http://activitystrea.ms/schema/1.0/person</type>
  <id>https://hubzilla.eu/channel/cb7f604332cf39/cb7f604332cf39</id>
  <title>cb7f604332cf39</title>
  <link rel="alternate" type="text/html" href="https://hubzilla.eu/channel/cb7f604332cf39"/>
  <link rel="photo" type="image/jpeg" href="https://f.haeder.net/photo/af3c7798d59aad5f28a889564d3d7bf6-5.png"/>
</object>
````

The body is very wrong, which prevented me from finding exactly where it was created. I feel like the issue is easy to fix, but I couldn't pinpoint the location where the name is used in place of the remote profile URL.

Any help is appreciated.

**Comment by rabuzarus** [comment]
This is related (or the same) to https://github.com/friendica/friendica/issues/3158

**Comment by Hypolite** [comment]
I feel it only is related, since it only concerns the display of such messages.

**Comment by MrPetovan** [comment]
I haven't seen such a post for a while, I'm passing for now.

**Comment by MrPetovan** [comment]
Another example: https://friendica.mrpetovan.com/display/adbba6c6185aa6f3402c5dd750369992

**Comment by rabuzarus** [comment]
It is is related to https://github.com/friendica/friendica/issues/3158

Problem is, that we work at different parts in the code with different formats for links.

old format:
```
<object>
  [...]
  <link>
    <link rel="alternate" type="text/html" href="https://hubzilla.eu/channel/cb7f604332cf39"/>
    <link rel="photo" type="image/jpeg" href="https://f.haeder.net/photo/af3c7798d59aad5f28a889564d3d7bf6-5.png"/>
  </link>
</object>
```

new format:
```
<object>
  [...]
  <link rel="alternate" type="text/html" href="https://hubzilla.eu/channel/cb7f604332cf39"/>
  <link rel="photo" type="image/jpeg" href="https://f.haeder.net/photo/af3c7798d59aad5f28a889564d3d7bf6-5.png"/>
</object>
```

We have never officially decided to use the "new" format. So it is more like a bug (which was introduced while rewriting the dfrn part two years ago). But personally I would vote for changing all links of our objects to the new format. It's a pity that the discussion about this topic in the helpers forum isn't available anymore.

The links for the diaspora connections (your issue) are constructed at:
https://github.com/friendica/friendica/blob/develop/src/Protocol/Diaspora.php#L2417 (and `Diaspora::constructNewFriendObject()`) It uses the new format.

while the construction of the connection message is done at:
https://github.com/friendica/friendica/blob/develop/include/conversation.php#L181
But `function localize_item()` expects the old format.

So we have two possibilities to solve this.
1.) Changing all links in objects to the old format (encapsulate all link elements with an separate link element)

2.) Change all links in objects to the new format (no encapsulating separate link element) and fix `function localize_item()` to read the new format.

Like I already said. I'm for the second solution.

**Comment by Quix0r** [comment]
Ah, the invalid link protocol and javascript:void(0) link you mean. :-)

**Comment by MrPetovan** [comment]
Yes!

**Comment by MrPetovan** [comment]
Closed via #5090
